blueocean-github-pipeline/src/main/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssue.java

```diff
@@ -60,26 +54,42 @@ public String getURL() {
                 return null;
             }
             MultiBranchProject mbp = (MultiBranchProject)job.getParent();
-            SCMSource source = Iterables.getFirst((List<SCMSource>)mbp.getSCMSources(), null);
-            if (!(source instanceof GitHubSCMSource)) {
+            SCMSource source = getFirst((List<SCMSource>)mbp.getSCMSources());
+            if (source == null || !(source instanceof GitHubSCMSource)) {
                 return null;
             }
             GitHubSCMSource gitHubSource = (GitHubSCMSource)source;
-            String apiUri = MoreObjects.firstNonNull(gitHubSource.getApiUri(), GitHubServerConfig.GITHUB_URL);
+            String apiUri = getApiUri(gitHubSource);
             final String repositoryUri = new HttpsRepositoryUriResolver().getRepositoryUri(apiUri, gitHubSource.getRepoOwner(), gitHubSource.getRepository());
-            return Collections2.transform(findIssueKeys(changeSetEntry.getMsg()), new Function<String, BlueIssue>() {
-                @Override
-                public BlueIssue apply(String input) {
-                    // Remove ".git"
-                    String uri = repositoryUri.substring(0, repositoryUri.length() - 4);
-                    return new GithubIssue("#" + input, String.format("%s/issues/%s", uri, input));
-                }
-            });
+            Collection<BlueIssue> results = new ArrayList<>();
+            for (String input : findIssueKeys(changeSetEntry.getMsg())) {
+                String uri = repositoryUri.substring(0, repositoryUri.length() - 4);
+                results.add(new GithubIssue("#" + input, String.format("%s/issues/%s", uri, input)));
+            }
+            return results;
+        }
+
+        private SCMSource getFirst(List<SCMSource> scmSources) {
+            if (scmSources.isEmpty()) {
+                return null;
+            } else {
+                return scmSources.get(0);
+            }
+        }
+
+    }
+
+    private static String getApiUri(GitHubSCMSource gitHubSource) {
```

Comment:
Nit: trinary 
```
String url = gitHubSource.getApiUri();
return url == null ? GitHubServerConfig.GITHUB_URL : url;
```

Reply:
I had trouble getting that past findbugs.. it may be smart enough for that though (it didn't recognise the get calls as idempotent)